### ISSUE TITLE
Issue with Block Properties

### ISSUE THREAD

**Issue by barrettj** [mysterioustrousers/MYSRuntime#1]
I'm getting the following warning output to console when running my app if I try to get the property list on a class which has a block as a property:
`*** -[__NSCFString substringWithRange:]: Range {2, 18446744073709551615} out of bounds; string length 2. This will become an exception for apps linked after 10.10 and iOS 8. Warning shown once per app execution.`

It appears that in `- (void)parseEncodingString:(NSString *)encodingString`

You're doing the following

```
        if ([encodingString length] > 1) {
            NSString *className = [encodingString substringWithRange:NSMakeRange(2, [encodingString length] - 3)];
            self.typeClass = NSClassFromString(className);
        }
```

Blocks have an encodingString of "@?" which will then cause the out of bounds message.

**Comment by atomkirk** [comment]
Ah interesting. Blocks are objects, hence the @ runtime type marker, but they don't have a public class. (I think it's private NSConcreteBlock or something).  I'll look into this. A sample project to play with would help tremendously

**Comment by barrettj** [comment]
https://dl.dropboxusercontent.com/u/882686/BlockPropertyTest.zip

**Comment by barrettj** [comment]
Any word on this?

**Comment by atomkirk** [comment]
I've added a reminder to my phone to look into it this evening

**Comment by barrettj** [comment]
Any luck?

**Comment by atomkirk** [comment]
Ok fixed in `0.0.5`
